File: clients/web/src/services/call.service.ts
import type { ApiResponse } from '../types'
import type { CallTokenResponse, CallType } from '../types/call'
import { http } from './http'

export const callService = {
  async start(conversationId: string, callType: CallType): Promise<ApiResponse<CallTokenResponse>> {
    const res = await http.post<ApiResponse<CallTokenResponse>>('/calls/start', {
      conversationId,
      callType,
    })
    return res.data
  },

  async answer(callId: string): Promise<ApiResponse<CallTokenResponse>> {
    const res = await http.post<ApiResponse<CallTokenResponse>>(`/calls/${callId}/answer`)
    return res.data
  },

  async decline(callId: string): Promise<ApiResponse<void>> {
    const res = await http.post<ApiResponse<void>>(`/calls/${callId}/decline`)
    return res.data
  },

  async end(callId: string): Promise<ApiResponse<void>> {
    const res = await http.post<ApiResponse<void>>(`/calls/${callId}/end`)
    return res.data
  },
}

File: clients/apps/lib/services/call_service.dart
import '../models/call.dart';
import '../network/call_api_client.dart';

class CallService {
  final CallApiClient _client;

  CallService(this._client);

  Future<CallTokenResponse> start(String conversationId, CallType callType) async {
    final res = await _client.startCall(
      StartCallBody(
        conversationId: conversationId,
        callType: callType.name,
      ),
    );
    return res.data.data!;
  }

  Future<CallTokenResponse> answer(String callId) async {
    final res = await _client.answerCall(callId);
    return res.data.data!;
  }

  Future<void> decline(String callId) async {
    await _client.declineCall(callId);
  }

  Future<void> end(String callId) async {
    await _client.endCall(callId);
  }
}
